src/nvim/message.c

```diff
@@ -1004,6 +1004,7 @@ void ex_messages(void *const eap_p)
       }
     }
     ui_call_msg_history_show(entries);
+    wait_return(false);
```

Comment:
we also want a `msg_history_clear()` event afterwards. adding it to `api/ui_events.in.h` and the `ui_call` here should be enough.